postgis/lwgeom_accum.c

```diff
@@ -114,64 +87,105 @@ pgis_geometry_accum_transfn(PG_FUNCTION_ARGS)
 
 	if ( PG_ARGISNULL(0) )
 	{
-		MemoryContext old = MemoryContextSwitchTo(aggcontext);
-		p = (pgis_abs*) palloc(sizeof(pgis_abs));
-		p->a = NULL;
-		p->data = (Datum) NULL;
-
-		if (PG_NARGS() == 3)
-		{
-			Datum argument = PG_GETARG_DATUM(2);
-			Oid dataOid = get_fn_expr_argtype(fcinfo->flinfo, 2);
+		int n = ((PG_NARGS()-2) <= CollectionBuildStateDataSize) ? (PG_NARGS()-2) : CollectionBuildStateDataSize;
```

Comment:
We'd only have `PG_NARGS()-2 >CollectionBuildStateDataSize` if someone improperly defined a SQL aggregate, right? Should it be an assert instead of a runtime thing that drops the extra args?

Reply:
yeah, it's guarding against future expansion that doesn't look at the underlying code. an assert could also do